lib/src/install/completion.rs

```diff
@@ -0,0 +1,298 @@
+    // Copy the host data into it (note this will follow symlinks; e.g.
+    // Anaconda in Fedora 41+ defaults to systemd-resolved)
+    proc1_root
+        .copy(RESOLVCONF, rootfs, RESOLVCONF)
+        .context("Copying new resolv.conf")?;
+    Ok(renamer)
+}
+
+/// Bind a mount point from the host namespace into our root
+fn bind_from_host(
+    rootfs: &Dir,
+    src: impl AsRef<Utf8Path>,
+    target: impl AsRef<Utf8Path>,
+) -> Result<()> {
+    fn bind_from_host_impl(rootfs: &Dir, src: &Utf8Path, target: &Utf8Path) -> Result<()> {
+        rootfs.create_dir_all(target)?;
+        if rootfs.is_mountpoint(target)?.unwrap_or_default() {
+            return Ok(());
+        }
+        let target = format!("/mnt/sysroot/{target}");
+        tracing::debug!("Binding {src} to {target}");
+        // We're run in a mount namespace, but not a pid namespace; use nsenter
+        // via the pid namespace to escape to the host's mount namespace and
+        // perform a mount there.
+        Command::new("nsenter")
+            .args(["-m", "-t", "1", "--", "mount", "--bind"])
+            .arg(src)
+            .arg(&target)
+            .run()?;
+        Ok(())
+    }
+
+    bind_from_host_impl(rootfs, src.as_ref(), target.as_ref())
+}
+
+/// Anaconda doesn't mount /sys/fs/cgroup in /mnt/sysroot
+#[context("Ensuring cgroupfs")]
+fn ensure_cgroupfs(rootfs: &Dir) -> Result<()> {
+    bind_from_host(rootfs, CGROUPFS, CGROUPFS)
+}
+
+/// If we have /etc/ostree/auth.json in the Anaconda environment then propagate
+/// it into /run/ostree/auth.json
+#[context("Propagating ostree auth")]
+fn ensure_ostree_auth(rootfs: &Dir, host_root: &Dir) -> Result<()> {
+    let Some((authpath, authfd)) =
+        ostree_ext::globals::get_global_authfile(&host_root).context("Querying authfiles")?
+    else {
+        tracing::debug!("No auth found in host");
+        return Ok(());
+    };
+    tracing::debug!("Discovered auth in host: {authpath}");
+    let mut authfd = io::BufReader::new(authfd);
+    let run_ostree_auth = Utf8Path::new(RUN_OSTREE_AUTH);
+    rootfs.create_dir_all(run_ostree_auth.parent().unwrap())?;
+    rootfs.atomic_replace_with(run_ostree_auth, |w| std::io::copy(&mut authfd, w))?;
+    Ok(())
+}
+
+#[context("Opening {PROC1_ROOT}")]
+fn open_proc1_root(rootfs: &Dir) -> Result<Dir> {
+    let proc1_root = rustix::fs::openat(
+        &rootfs.as_fd(),
+        PROC1_ROOT,
+        OFlags::CLOEXEC | OFlags::DIRECTORY,
+        Mode::empty(),
+    )?;
+    Dir::reopen_dir(&proc1_root.as_fd()).map_err(Into::into)
+}
+
+/// Core entrypoint invoked when we are likely being invoked from inside Anaconda as a `%post`.
+pub(crate) async fn run_from_anaconda(rootfs: &Dir) -> Result<()> {
+    // unshare our mount namespace, so any *further* mounts aren't leaked.
+    // Note that because this does a re-exec, anything *before* this point
+    // should be idempotent.
+    crate::cli::ensure_self_unshared_mount_namespace()?;
```

Comment:
Requires root, and will clumsily fail. Since this is what people will see if they run `bootc install ensure-completion` (e.g. out of curiosity while digging into our anaconda procedures), maybe we should error out more gracefully?